cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -177,32 +177,30 @@ protected String getServiceName() {
 
   @Override
   protected void startUp() throws Exception {
-    final Job job = createJob();
-    Configuration mapredConf = job.getConfiguration();
+    // Creates a temporary directory locally for storing all generated files.
+    File tempDir = createTempDirectory();
+    this.cleanupTask = createCleanupTask(tempDir);
 
-    classLoader = new MapReduceClassLoader(context.getProgram().getClassLoader());
-    mapredConf.setClassLoader(new WeakReferenceDelegatorClassLoader(classLoader));
-    ClassLoaders.setContextClassLoader(mapredConf.getClassLoader());
+    try {
+      Job job = createJob(new File(tempDir, "mapreduce"));
```

Comment:
``programType`` is already part of the ``tempDir``, do we need a subdirectory called "mapreduce" too?